omero/examples/java/JSONClient.java

```diff
@@ -0,0 +1,484 @@
+     * @return The response
+     * @throws HttpException
+     *             If something went wrong
+     * @throws ClientProtocolException
+     *             If something went wrong
+     * @throws IOException
+     *             If something went wrong
+     */
+    private JsonStructure put(String url, JsonObject data)
+            throws HttpException, ClientProtocolException, IOException {
+        HttpPut httpPut = new HttpPut(url);
+        if (data != null) {
+            StringEntity requestEntity = new StringEntity(data.toString(),
+                    ContentType.APPLICATION_JSON);
+            httpPut.setEntity(requestEntity);
+        }
+        httpPut.addHeader("X-CSRFToken", this.token);
+        HttpResponse res = httpClient.execute(httpPut);
+        if (res.getStatusLine().getStatusCode() != 200)
+            throw new HttpException("PUT failed. URL: " + url + " Status:"
+                    + res.getStatusLine());
+
+        try (JsonReader reader = Json.createReader(new BufferedReader(
+                new InputStreamReader(res.getEntity().getContent())))) {
+            return reader.read();
+        }
+    }
+
+    /**
+     * Perform a POST request
+     * 
+     * @param url
+     *            The request URL
+     * @param params
+     *            The paramters
+     * @return The response
+     * @throws HttpException
+     *             If something went wrong
+     * @throws ClientProtocolException
+     *             If something went wrong
+     * @throws IOException
+     *             If something went wrong
+     */
+    private JsonStructure post(String url, Map<String, String> params)
+            throws HttpException, ClientProtocolException, IOException {
+
+        HttpPost httpPost = new HttpPost(url);
+        if (params != null && !params.isEmpty()) {
+            List<NameValuePair> nvps = new ArrayList<NameValuePair>();
+            for (Entry<String, String> entry : params.entrySet()) {
+                nvps.add(new BasicNameValuePair(entry.getKey(), entry
+                        .getValue()));
+            }
+            httpPost.setEntity(new UrlEncodedFormEntity(nvps));
+        }
+        httpPost.addHeader("X-CSRFToken", this.token);
+        HttpResponse res = httpClient.execute(httpPost);
+        if (res.getStatusLine().getStatusCode() != 200)
+            throw new HttpException("POST failed. URL: " + url + " Status:"
+                    + res.getStatusLine());
+
+        try (JsonReader reader = Json.createReader(new BufferedReader(
+                new InputStreamReader(res.getEntity().getContent())))) {
+            return reader.read();
+        }
+    }
+    
+    public static void main(String[] args) throws Exception {
+        String baseURL = "";
+        String username = "";
+        String password = "";
+        String servername = "";
+
+        for (int i = 0; i < args.length; i++) {
+            try {
+                if (args[i].trim().startsWith("--omero.webhost")) {
```

Comment:
I'll fix it later when the JSONClient example gets moved to `training/examples`, where it'll be run by the CI training job.